server/mediaserver.go

```diff
@@ -616,19 +630,23 @@ func (s *LivepeerServer) HandlePush(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		ticker := time.NewTicker(RefreshIntervalHttpPush)
+		ticker := time.NewTicker(refreshIntervalHttpPush)
 
 		go func(s *LivepeerServer, mid core.ManifestID) {
+			defer ticker.Stop()
 			for range ticker.C {
 				s.connectionLock.RLock()
-				lastUsed := s.rtmpConnections[mid].lastUsed
-				s.connectionLock.RUnlock()
```

Comment:
Best to keep lock handling to a minimum since it's so easy to mess up. Right now, the unlock needs to be special-cased in three different places, which makes reasoning about the flow difficult. An improvement would be something like:

```golang
lastUsed := time.Now().Sub(2 * refreshIntervalHttpPush) // default to remove if nonexistent
s.connectionLock.RLock()
cxn, exists := s.rtmpConnections[mid]
if exists {
  lastUsed = cxn.lastUsed
}
s.connectionLock.RULock()

if time.Since(lastUsed) {
  removeRTMPStream(s, mid)
}
```

Reply:
Fixed